packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,103 @@
         default:
           return 'navbar-dark bg-primary'
         }
-      }
+      },
     },
     methods: {
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url.toLowerCase();
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`.toLowerCase();
+        } else if (!url.endsWith('/')) {
+          return `${url}.html`.toLowerCase();
+        } else {
+          return url.toLowerCase();
+        }
+      },
+      // Splits a normalised URL into its parts, e.g http://site.org/foo/bar/index.html -> ['foo','bar','index.html']
+      splitUrl(url) {
+        const u = new URL(this.normalizeUrl(url));
+        return `${u.pathname}`.substr(1).split('/');
+      },
+      isSibling(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (hParts.length !== uParts.length) {
+          return false;
+        }
+        for (let i = 0; i < hParts.length - 1; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isChild(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (hParts.length <= uParts.length) {
```

Comment:
should this be reversed? (since we're checking whether the page you are at (url) is a _child_ of the link (href))

Reply:
Hmm could I ask why?

Reply:
Hmm ok, I think the implementation I had in mind was different; checking if href was a child of url (which was incorrect)